src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1672,36 +1672,55 @@ void UpdateNavigationBarVisibility(bool animated)
 
 			void UpdateToolbarItems()
 			{
-				if (NavigationItem.RightBarButtonItems != null)
+				if (NavigationItem.RightBarButtonItems is not null)
 				{
 					for (var i = 0; i < NavigationItem.RightBarButtonItems.Length; i++)
+					{
 						NavigationItem.RightBarButtonItems[i].Dispose();
+					}
 				}
-				if (ToolbarItems != null)
+
+				if (ToolbarItems is not null)
 				{
 					for (var i = 0; i < ToolbarItems.Length; i++)
+					{
 						ToolbarItems[i].Dispose();
+					}
 				}
 
 				List<UIBarButtonItem> primaries = null;
-				List<UIBarButtonItem> secondaries = null;
+				List<UIMenuElement> secondaries = null;
 				var toolbarItems = _tracker.ToolbarItems;
 				foreach (var item in toolbarItems)
 				{
 					if (item.Order == ToolbarItemOrder.Secondary)
-						(secondaries = secondaries ?? new List<UIBarButtonItem>()).Add(item.ToUIBarButtonItem(true));
+					{
+						(secondaries ??= []).Add(item.ToUIMenuElement());
+					}
 					else
-						(primaries = primaries ?? new List<UIBarButtonItem>()).Add(item.ToUIBarButtonItem());
+					{
+						(primaries ??= []).Add(item.ToUIBarButtonItem());
+					}
 				}
 
-				if (primaries != null)
+				if (primaries is not null)
+				{
 					primaries.Reverse();
-				NavigationItem.SetRightBarButtonItems(primaries == null ? Array.Empty<UIBarButtonItem>() : primaries.ToArray(), false);
-				ToolbarItems = secondaries == null ? Array.Empty<UIBarButtonItem>() : secondaries.ToArray();
+				}
 
-				NavigationRenderer n;
-				if (_navigation.TryGetTarget(out n))
+				if (secondaries is not null && secondaries.Count > 0)
+				{
+					var menu = UIMenu.Create(string.Empty, null, UIMenuIdentifier.Edit, UIMenuOptions.DisplayInline, secondaries.ToArray());
+					var menuButton = new UIBarButtonItem(UIImage.GetSystemImage("ellipsis.circle"), menu);
```

Comment:
Could be nice a comment here pointing to the documentation https://developer.apple.com/design/human-interface-guidelines/pull-down-buttons about recommendations and using the ellipsis.circle symbol here.